src/transactions/feed/JumpstartTransactionDetailsScreen.tsx

```diff
@@ -0,0 +1,320 @@
+    abi: walletJumpstart.abi,
+    eventName: ['ERC20Deposited'],
+    logs: transactionReceipt.logs,
+  })
+
+  if (parsedLogs.length != 1) {
+    throw new Error('Unexpected number of matching logs')
+  }
+
+  const { beneficiary, index } = parsedLogs[0].args
+
+  Logger.debug('Decoded event', { beneficiary, index })
+
+  const erc20Claim = await viemClient.readContract({
+    address: jumpstartContractAddress as Address,
+    abi: walletJumpstart.abi,
+    functionName: 'erc20Claims',
+    args: [beneficiary, index],
+  })
+
+  const claimed = erc20Claim[3]
+
+  Logger.debug(`Reward deposited in ${transactionHash} ${claimed ? 'was' : 'was NOT'} claimed`)
+
+  return { beneficiary, index: Number(index), claimed }
+}
+
+function JumpstartTransactionDetailsScreen({ route }: Props) {
+  const { transaction } = route.params
+  const { t } = useTranslation()
+
+  const [reclaimTx, setReclaimTx] = useState<SerializableTransactionRequest | null>(null)
+  const transactionString = useMemo(() => (reclaimTx ? JSON.stringify(reclaimTx) : ''), [reclaimTx])
+
+  const transactionTokenInfo = useTokenInfo(transaction.amount.tokenId)
+  const parsedAmount = new BigNumber(transaction.amount.value).abs()
+  const token = useTokenInfo(transaction.amount.tokenId)
+  const isDeposit = transaction.type === TokenTransactionTypeV2.Sent
+  const jumpstartContractAddress = transaction.address
+
+  const walletAddress = useSelector(walletAddressSelector)
+  const bottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const title =
+    transaction.type === TokenTransactionTypeV2.Sent
+      ? t('feedItemJumpstartTitle')
+      : t('feedItemJumpstartReceivedSubtitle')
+
+  const fetchClaimData = useAsync(
+    async () => {
+      if (!isDeposit) {
+        return
+      }
+      return await getClaimDataAndStatus(
+        jumpstartContractAddress as Address,
+        transaction.networkId,
+        transaction.transactionHash as Hash
+      )
+    },
+    [],
+    {
+      onError: (error) => {
+        // TODO: Show error message in the UI
+        Logger.error(TAG, 'Failed to fetch escrow data', error)
+      },
+    }
+  )
+
+  if (!token) {
+    // should never happen
+    Logger.error(TAG, 'Token is undefined')
+    return null
+  }
+
+  const onReclaimPress = () => {
+    if (!fetchClaimData.result) {
+      Logger.error(TAG, 'No escrow data found when trying to reclaim')
+      return
+    }
+
+    const { beneficiary, index } = fetchClaimData.result
+
+    setReclaimTx({
+      from: walletAddress as Address,
+      to: jumpstartContractAddress as Address,
+      value: '0',
+      data: encodeFunctionData({
+        abi: walletJumpstart.abi,
+        functionName: 'reclaimERC20',
+        args: [beneficiary, BigInt(index)],
+      }),
+      // TODO: Estimate
+      gas: '130000',
+      maxFeePerGas: '1000000000',
+    })
+
+    bottomSheetRef.current?.snapToIndex(0)
+  }
+
+  const isClaimed = fetchClaimData.result?.claimed
+
+  const reclaimButton = (
```

Comment:
this is personal preference but i think this can go directly into the render rather than be a variable, it would be easier to read all the render logic in one place

Reply:
Yeah, I think I prefer to keep the render tree as simple as I can. But I lost that fight in this repo, so I should stick to the standard, this is simple enough to be there. 👍